pkg/pagerduty/syncset.go

```diff
@@ -34,7 +34,7 @@ func (data *Data) GenerateSyncSet(namespace string, name string, pdServiceID str
 		Spec: hivev1.SyncSetSpec{
 			ClusterDeploymentRefs: []corev1.LocalObjectReference{
 				{
-					Name: name,
+					Name: data.ClusterID,
```

Comment:
Think the `ClusterDeploymentRefs` value needs to be metadata.name, which is passed as `name` param to this function.  The ClusterID is being set to the `ClusterID` property in the ClusterDeployment spec.